Doc/library/email.utils.rst

```diff
@@ -124,7 +124,10 @@ of the new API.
 .. function:: parsedate_to_datetime(date)
 
    The inverse of :func:`format_datetime`.  Performs the same function as
-   :func:`parsedate`, but on success returns a :mod:`~datetime.datetime`.  If
+   :func:`parsedate`, but on success returns a :mod:`~datetime.datetime`;
```

Comment:
I'm looking at this now during the Python core sprint.  Thanks for resurrecting this PR.

I find this description a bit confusing because it's saying that the function either returns `None` or raises `ValueError` when parsing an invalid date.  That makes it difficult to know when I'll get `None` and when I'll get an exception.  I think you're saying that only some bogus strings will raise `ValueError`, e.g. those where the hour is invalid.  I also don't see where the tests you're adding include the test for `None` return.

I'm not sure what to suggest, because I'm not sure what the intended semantics *should* be.

Reply:
I actually don't see how *parsedate_to_datetime* would ever return None. It is *_parsedate_tz* which returns **None** on bogus input, in which case *parsedate_to_datetime* raises a **TypeError**.
This is also covered in the tests, so those should be fine.

In order to continue I suggest to fix the documentation on *parsedate_to_datetime*, remove the mention of it returning None and replacing it with it possibly returning **TypeError** in case of an invalid date input.

Reply:
As I mentioned in the ticket, I agree with you; `parsedate_to_datetime()` can't return None, so let's remove that from the documentation.  I also suggest checking for `None` return from `_parsedate_tz()` and turning that into an explicit raise of `ValueError` rather than relying on the implicit `TypeError` from a failing tuple unpack.  Then the docs should say that `ValueError` is raised on an invalid date input.

Reply:
Thanks for your comments. I fully agree with your points, that makes it definitely cleaner.
I implemented them in my latest commit.